master/internal/api_resourcepool.go

```diff
@@ -31,7 +36,28 @@ func (a *apiServer) OverwriteRPWorkspaceBindings(
 func (a *apiServer) UnbindRPFromWorkspace(
 	ctx context.Context, req *apiv1.UnbindRPFromWorkspaceRequest,
 ) (*apiv1.UnbindRPFromWorkspaceResponse, error) {
-	return nil, nil
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+	// Check permissions for all workspaces. Return err if any workspace doesn't have permissions.
+	// No partial unbinding.
+	for _, workspaceID := range req.WorkspaceIds {
+		if err = workspaceauth.AuthZProvider.Get().CanUnBindRPWorkspace(ctx, *curUser,
```

Comment:
hmm I think I would put the for loop inside of the authZ function instead of in the api handler, mainly because the permission checking action is a database query and we could do a bulk query by passing thru the list of workspaceIDs

Reply:
oops yes true